src/loaders/workspace_uploader/workspace_uploader.py

```
@@ -332,115 +357,99 @@ def _post_process(
 
     Parameters:
-        conf: Conf object
+        asu: AssemblyUtil client
+        ws: Workspace client
         upload_env_key: environment variable key in uploaded.yaml file
         workspace_id: target workspace id
         upload_dir: a directory in collectionssource that creates new directories linking to sourcedata
         wait_to_upload_assemblies: a dictionary that maps assembly file name to assembly directory
+        batch_size: a number of files to upload per batch
+        output_dir: a directory in sourcedata/workspace to store new assembly entries
 
     Returns:
-        a list of assembly names that failed to upload
+        number of assembly files have been sucessfully uploaded from wait_to_upload_assemblies
     """
     assembly_files_len = len(wait_to_upload_assemblies)
-    print(f"Start uploading {assembly_files_len} assembly files with {conf.workers} workers\n")
+    print(f"Start uploading {assembly_files_len} assembly files\n")
 
-    # Put the assembly files in the input_queue
-    counter = 1
-    for assembly_name, host_assembly_dir in wait_to_upload_assemblies.items():
+    uploaded_count = 0
+    uploaded_fail = False
+    for assembly_tuples in _gen(wait_to_upload_assemblies, batch_size):
 
-        container_internal_assembly_path = os.path.join(
-            JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, assembly_name
-        )
-        conf.input_queue.put(
-            (
+        try:
+            batch_upas = _upload_assemblies_to_workspace(asu, workspace_id, assembly_tuples)
+            batch_uploaded_tuples = assembly_tuples
+        except Exception as e:
+            name_assemblyTuple_map = {
+                assembly_tuple.assembly_name: assembly_tuple for assembly_tuple in assembly_tuples
+            }
+            assembly_objects_in_ws = loader_helper.list_objects(
+                workspace_id,
+                ws,
+                loader_common_names.OBJECTS_NAME_ASSEMBLY,
```

Rather than listing objects here, which lists every assembly in the workspace and so could list and return hundreds of thousands of objects for GTDB, I would use `get_object_info3` with ignore errors set to true and only submit the names for the current batch.

---

Updated!

---

:+1: